indent/cylc.vim

```diff
@@ -0,0 +1,68 @@
+" Only load this indent file when no other was loaded.
+if exists("b:did_indent")
+  finish
+endif
+let b:did_indent = 1
+
+" Some preliminary settings
+setlocal nolisp      " Make sure lisp indenting doesn't supersede us
+setlocal autoindent  " indentexpr isn't much help otherwise
+
+setlocal indentexpr=GetCylcIndent(v:lnum)
+setlocal indentkeys=!^F,o,O,]
+
+" Only define the function once.
+if exists("*GetCylcIndent")
+  finish
+endif
+
+let s:save_cpo = &cpo
+set cpo&vim
+
+function s:header_level(lnum)
+  " Level of the section header on a given line, 0 if none
+  let line = getline(a:lnum)
+  return substitute(line, '^\s*\(\[*\).*', '\=strlen(submatch(1))', '')
+endfunction
+
+function s:prev_header(lnum)
+  " Jump to the header preceding a given line
+  call cursor(a:lnum, 1)
+  return search('^\s*\[\+', "bW")
+endfunction
+
+function GetCylcIndent(lnum)
+  let line = getline(a:lnum)
+  let line = substitute(line, '\v^\s+|\s+$', '', 'g')
+
+  " Do not adjust jinja indentation
+  if line =~ '^\({{|{%\)'
```

Comment:
I think this would add support for the EmPy syntaxes:

```
@{
@[
@(
```

```suggestion
  if line =~ '^\({{|{%|@\[|@{|@(\)'
```

Reply:
Thanks, I admittedly overlooked empy as I'm not personally familiar.

Made me realise that the pattern was wrong to start with - `|` is actually a literal bar while `\|` separates alternatives.